### ISSUE TITLE
Speed up stale image checker

### ISSUE THREAD

**Issue by Eric-Arellano** [Qiskit/documentation#2530]
https://github.com/Qiskit/documentation/pull/2528 improved the stale image checker to be "fast enough" to run occasionally, but the program is still quite slow due to starting up ripgrep 3450 times. It takes several minutes on my mac.

I suspect a faster approach would be to read the markdown to extract all used images, such as by using the link checker, then compare that to the images we do have on disk. It would avoid subprocess calls. However, it's possible this solution would overwhelm system memory.

**Comment by arnaucasau** [comment]
What do you think about integrating the stale image checker with our alt text image checker in https://github.com/Qiskit/documentation/blob/main/scripts/js/commands/checkImages.ts? I think we could use the unified plugin to find the stale images:

https://github.com/Qiskit/documentation/blob/be6687be304ffc3262c6c21bb6117d21531821f6/scripts/js/lib/markdownImages.ts#L31-L39

**Comment by Eric-Arellano** [comment]
That might be helpful to use the stale image checker. However, the stale image checker doesn't check historical API docs. Also, I realize we could probably implement https://github.com/Qiskit/documentation/issues/2538 such that we no longer even need to worry about checking API docs. If true, that would speed up this overall program enough automatically that we could avoid spending more dev resources on this ticket.

**Comment by Eric-Arellano** [comment]
Indeed, https://github.com/Qiskit/documentation/pull/2541 fixes the performance problem 🚀

It's blocked by https://github.com/Qiskit/documentation/issues/2540, which I want to fix because that is a total gotcha for future contributors.
